src/components/Graphs/SalesGraph.vue

```diff
@@ -0,0 +1,119 @@
+<script setup lang="ts">
+import { onMounted, PropType, ref, watch } from 'vue';
+import Chart from 'chart.js/auto';
+
+import { completeMonths } from '@/utils/timeData';
+import Card from '@/components/Card.vue';
+
+import type { SalesDataItem } from '@/interfaces/sales.interface';
+
+const props = defineProps({
+  salesData: {
+    type: Array as PropType<SalesDataItem[]>,
+    required: true,
+  }
+});
+
+const salesChart = ref<Chart<'line', number[], string> | null>(null);
+
+function initializeChart(): void {
+  const ctx = document.getElementById('salesChart') as HTMLCanvasElement;
+  if (!ctx) return;
+
+  // Crear un mapa para facilitar el acceso a los datos proporcionados
+  const dataMap = new Map<string, number>(
+    props.salesData.map(item => [item.month, item.revenue])
+  );
+
+  // Preparar las etiquetas (labels) y los datos (data) completos
+  const labels: string[] = [];
+  const data: number[] = [];
+
+  // Iterar sobre todos los meses completos
+  completeMonths.forEach(month => {
+    labels.push(`${month} 2024`);  // Asumimos que el año es siempre 2024
```

Comment:
the year not always will be 2024, use typescript to get the year, or if you promise me that all years January 1st at 12:01 am you will change the year to the correct year, we can let that as you did